services/analyze-sequelize-tables.js

```diff
@@ -62,100 +62,210 @@ function hasTimestamps(fields) {
   return hasCreatedAt && hasUpdatedAt;
 }
 
-function formatAliasName(columnName) {
-  const alias = _.camelCase(columnName);
-  if (alias.endsWith('Id') && alias.length > 2) {
-    return alias.substring(0, alias.length - 2);
-  }
-  if (alias.endsWith('Uuid') && alias.length > 4) {
-    return alias.substring(0, alias.length - 4);
-  }
-  return alias;
-}
-
 // NOTICE: Look for the id column in both fields and primary keys.
 function hasIdColumn(fields, primaryKeys) {
   return fields.some((field) => field.name === 'id' || field.nameColumn === 'id')
     || _.includes(primaryKeys, 'id');
 }
 
-function analyzeTable(table, config) {
+// NOTICE: Add isComposite, isUnique and isPrimary attributes to the foreign keys
+async function setForeignKeys(schema, constraints, primaryKeys) {
+  const foreignKeys = [];
+  for (let index = 0; index < constraints.length; index += 1) {
+    const foreignKey = constraints[index];
+    const columnName = foreignKey.column_name;
+    let uniqueIndexes = foreignKey.unique_indexes || null;
+
+    // NOTICE: mssql doesn't support aggregation to JSON, we need to parse it
+    if (uniqueIndexes !== null && uniqueIndexes === 'string') {
+      uniqueIndexes = JSON.parse(uniqueIndexes);
+    }
+
+    if (foreignKey.column_type === 'FOREIGN KEY') {
+      const hasUniqueConstraintFromIndex = uniqueIndexes !== null
+        && uniqueIndexes.length === 1
+        && _.find(uniqueIndexes, (array) => {
+          if (array.length === 1) return array.includes(columnName);
+          return false;
+        }) !== undefined;
+
+      foreignKey.isInCompositeKey = primaryKeys
+        && primaryKeys.length > 1
+        && primaryKeys.includes(columnName);
+
+      foreignKey.isUnique = _.find(constraints, { column_name: columnName, column_type: 'UNIQUE' }) !== undefined
+        || hasUniqueConstraintFromIndex;
+
+      foreignKey.isPrimary = _.isEqual([columnName], primaryKeys);
```

Comment:
A composite key is not the only primaryKey

Reply:
If I tag a column that is part of a composite primary key as a primary key, then it will be added as a hasOne (and I don't need the isPrimary to be true on it to establish if its part of many to many anyway)